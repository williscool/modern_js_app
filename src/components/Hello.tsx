import * as React from "react";

export interface Props {
  name: string;
  enthusiasmLevel?: number;
}

function Hello({ name, enthusiasmLevel = 1 }: Props) {
  if (enthusiasmLevel <= 0) {
    throw new Error("You could be a little more enthusiastic. :D");
  }

  return (
    <div className="hello">
      <h1 className="greeting">
        Hello from {name + getExclamationMarks(enthusiasmLevel)}
      </h1>
    </div>
  );
}

// helpers

function getExclamationMarks(numChars: number) {
  return Array(numChars + 1).join("!");
}

export default Hello;
