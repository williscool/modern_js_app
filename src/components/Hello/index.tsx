import * as React from "react";

// https://github.com/Microsoft/tslint-microsoft-contrib/issues/339
// tslint:disable:react-unused-props-and-state
export interface Props {
  name: string;
  enthusiasmLevel?: number;
}
// tslint:enable:react-unused-props-and-state

/**
 * Say Hello To My little friend
 *
 * @param {Props} { name, enthusiasmLevel = 1 }
 * @returns {JSX.Element}
 */
// tslint:disable-next-line:function-name
export function Hello({ name, enthusiasmLevel = 1 }: Props): JSX.Element {
  if (enthusiasmLevel <= 0) {
    throw new Error("You could be a little more enthusiastic. :D");
  }

  return (
    <div className="hello">
      <p className="greeting">
        Hello from {name + getExclamationMarks(enthusiasmLevel)}
      </p>
    </div>
  );
}

// helpers

/**
 * Get the number of exclamations!
 *
 * so excited!
 *
 * @param {number} numChars
 * @returns {string}
 */
function getExclamationMarks(numChars: number): string {
  return Array(numChars + 1).join("!");
}
