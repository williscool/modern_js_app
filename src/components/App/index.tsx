import * as React from "react";
import { Hello } from "../Hello";

/**
 * Entry Point of the application
 *
 * @export
 * @function App
 * @returns {React.SFC}
 */
// tslint:disable:variable-name
export const App: React.SFC = props => (
  <div className="App">
    <Hello name="Webpack and TypeScript" enthusiasmLevel={25} />,
  </div>
);
