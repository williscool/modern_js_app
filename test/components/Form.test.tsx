// src/components/Hello.test.tsx

/**
 * Tests form
 *
 * Testing the overall sturcture of the form would be covered by the snapshot test
 *
 * So no need to test like where the actions are in the form or if they are all listed.
 *
 * But there is special logic to hide a currency based on if its in the quote or base currency field
 *
 * I think that is worth testing.
 *
 * Also I will seperately test the validation logic in more of a unit test as well
 *
 */

import * as Enzyme from "enzyme";
import * as enzymeAdapterReact16 from "enzyme-adapter-react-16";
import MenuItem from "material-ui/MenuItem";
import * as React from "react";
import { Curriences, Form } from "../../src/components/Form";

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

it("renders without error period", () => {
  const form = Enzyme.shallow(<Form />);
  expect(form.find("form")).toBeDefined();
});

it("renders correctly", () => {
  const form = Enzyme.shallow(<Form />);
  expect(form).toMatchSnapshot();
});

it("renders correct menu items for currency selectfields", () => {
  // why this doesn't work as expected https://github.com/airbnb/enzyme/issues/252
  // left for the spirit of whats here

  const form = Enzyme.shallow(<Form />);
  // const numCurrencyTypes = Object.keys(Curriences).length;
  const baseSelect = form.find(".base");
  const quoteSelect = form.find(".quote");

  const baseMenuItems = baseSelect.find(<MenuItem />);
  const quoteMenuItems = quoteSelect.find(<MenuItem />);

  // expect one less menuitem than there are currencies
  // expect(baseMenuItems.length).toBe(numCurrencyTypes - 1);
  // expect(quoteMenuItems.length).toBe(numCurrencyTypes - 1);

  // On load BTC is the default quote currency.
  // So we want to test that the base select CANNOT BE set to BTC
  expect(
    baseMenuItems.filter((mi: MenuItem) => mi.props.value === Curriences.BTC)
      .length
  ).toBe(0);

  // On load USD is the default base currency.
  // So we want to test that the quote select CANNOT BE set to USD
  expect(
    quoteMenuItems.filter((mi: MenuItem) => mi.props.value === Curriences.USD)
      .length
  ).toBe(0);

  // you could also test that simulating a change properly updates the menu items as well
});
