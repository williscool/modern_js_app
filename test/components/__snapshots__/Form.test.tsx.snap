// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SelectField
          autoWidth={false}
          className="action"
          disabled={false}
          errorText=""
          floatingLabelFixed={true}
          floatingLabelText="Action"
          fullWidth={false}
          multiple={false}
          onChange={[Function]}
          value="Buy"
        >
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="Buy"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="Buy"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="Sell"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="Sell"
          />
        </SelectField>,
        <br />,
        <SelectField
          autoWidth={false}
          className="base"
          disabled={false}
          errorText=""
          floatingLabelFixed={true}
          floatingLabelText="Base Currency"
          fullWidth={false}
          multiple={false}
          onChange={[Function]}
          value="USD"
        >
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="USD"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="USD"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="LTC"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="LTC"
          />
        </SelectField>,
        <br />,
        <SelectField
          autoWidth={false}
          className="quote"
          disabled={false}
          errorText=""
          floatingLabelFixed={true}
          floatingLabelText="Quote Currency"
          fullWidth={false}
          multiple={false}
          onChange={[Function]}
          value="BTC"
        >
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="BTC"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="BTC"
          />
          <MenuItem
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="LTC"
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            value="LTC"
          />
        </SelectField>,
        <br />,
        <TextField
          className="amount"
          disabled={false}
          errorText=""
          floatingLabelFixed={true}
          floatingLabelText="Amount"
          fullWidth={false}
          hintText="10.00"
          multiLine={false}
          name="amount"
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
          value=""
        />,
        <br />,
        <br />,
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Submit"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false}
        />,
        <p>
           
          {
  "actions": [
    "Buy",
    "Sell"
  ],
  "current_action": "Buy",
  "curriences": [
    "USD",
    "BTC",
    "LTC"
  ],
  "base": "USD",
  "quote": "BTC",
  "amount": "",
  "actionErrorText": "",
  "baseErrorText": "",
  "quoteErrorText": "",
  "amountErrorText": ""
}
           
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoWidth": false,
          "children": Array [
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Buy"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="Buy"
            />,
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Sell"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="Sell"
            />,
          ],
          "className": "action",
          "disabled": false,
          "errorText": "",
          "floatingLabelFixed": true,
          "floatingLabelText": "Action",
          "fullWidth": false,
          "multiple": false,
          "onChange": [Function],
          "value": "Buy",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "Buy",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "Buy",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "Sell",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "Sell",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoWidth": false,
          "children": Array [
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="USD"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="USD"
            />,
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="LTC"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="LTC"
            />,
          ],
          "className": "base",
          "disabled": false,
          "errorText": "",
          "floatingLabelFixed": true,
          "floatingLabelText": "Base Currency",
          "fullWidth": false,
          "multiple": false,
          "onChange": [Function],
          "value": "USD",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "USD",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "USD",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "LTC",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "LTC",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoWidth": false,
          "children": Array [
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="BTC"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="BTC"
            />,
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="LTC"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="LTC"
            />,
          ],
          "className": "quote",
          "disabled": false,
          "errorText": "",
          "floatingLabelFixed": true,
          "floatingLabelText": "Quote Currency",
          "fullWidth": false,
          "multiple": false,
          "onChange": [Function],
          "value": "BTC",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "BTC",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "BTC",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "LTC",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "LTC",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "amount",
          "disabled": false,
          "errorText": "",
          "floatingLabelFixed": true,
          "floatingLabelText": "Amount",
          "fullWidth": false,
          "hintText": "10.00",
          "multiLine": false,
          "name": "amount",
          "onChange": [Function],
          "rows": 1,
          "type": "text",
          "underlineShow": true,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "fullWidth": false,
          "label": "Submit",
          "labelPosition": "after",
          "onClick": [Function],
          "primary": true,
          "secondary": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "{
  \\"actions\\": [
    \\"Buy\\",
    \\"Sell\\"
  ],
  \\"current_action\\": \\"Buy\\",
  \\"curriences\\": [
    \\"USD\\",
    \\"BTC\\",
    \\"LTC\\"
  ],
  \\"base\\": \\"USD\\",
  \\"quote\\": \\"BTC\\",
  \\"amount\\": \\"\\",
  \\"actionErrorText\\": \\"\\",
  \\"baseErrorText\\": \\"\\",
  \\"quoteErrorText\\": \\"\\",
  \\"amountErrorText\\": \\"\\"
}",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "{
  \\"actions\\": [
    \\"Buy\\",
    \\"Sell\\"
  ],
  \\"current_action\\": \\"Buy\\",
  \\"curriences\\": [
    \\"USD\\",
    \\"BTC\\",
    \\"LTC\\"
  ],
  \\"base\\": \\"USD\\",
  \\"quote\\": \\"BTC\\",
  \\"amount\\": \\"\\",
  \\"actionErrorText\\": \\"\\",
  \\"baseErrorText\\": \\"\\",
  \\"quoteErrorText\\": \\"\\",
  \\"amountErrorText\\": \\"\\"
}",
          " ",
        ],
        "type": "p",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SelectField
            autoWidth={false}
            className="action"
            disabled={false}
            errorText=""
            floatingLabelFixed={true}
            floatingLabelText="Action"
            fullWidth={false}
            multiple={false}
            onChange={[Function]}
            value="Buy"
          >
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Buy"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="Buy"
            />
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Sell"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="Sell"
            />
          </SelectField>,
          <br />,
          <SelectField
            autoWidth={false}
            className="base"
            disabled={false}
            errorText=""
            floatingLabelFixed={true}
            floatingLabelText="Base Currency"
            fullWidth={false}
            multiple={false}
            onChange={[Function]}
            value="USD"
          >
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="USD"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="USD"
            />
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="LTC"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="LTC"
            />
          </SelectField>,
          <br />,
          <SelectField
            autoWidth={false}
            className="quote"
            disabled={false}
            errorText=""
            floatingLabelFixed={true}
            floatingLabelText="Quote Currency"
            fullWidth={false}
            multiple={false}
            onChange={[Function]}
            value="BTC"
          >
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="BTC"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="BTC"
            />
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="LTC"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
              value="LTC"
            />
          </SelectField>,
          <br />,
          <TextField
            className="amount"
            disabled={false}
            errorText=""
            floatingLabelFixed={true}
            floatingLabelText="Amount"
            fullWidth={false}
            hintText="10.00"
            multiLine={false}
            name="amount"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
          />,
          <br />,
          <br />,
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Submit"
            labelPosition="after"
            onClick={[Function]}
            primary={true}
            secondary={false}
          />,
          <p>
             
            {
  "actions": [
    "Buy",
    "Sell"
  ],
  "current_action": "Buy",
  "curriences": [
    "USD",
    "BTC",
    "LTC"
  ],
  "base": "USD",
  "quote": "BTC",
  "amount": "",
  "actionErrorText": "",
  "baseErrorText": "",
  "quoteErrorText": "",
  "amountErrorText": ""
}
             
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoWidth": false,
            "children": Array [
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Buy"
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                value="Buy"
              />,
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="Sell"
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                value="Sell"
              />,
            ],
            "className": "action",
            "disabled": false,
            "errorText": "",
            "floatingLabelFixed": true,
            "floatingLabelText": "Action",
            "fullWidth": false,
            "multiple": false,
            "onChange": [Function],
            "value": "Buy",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "Buy",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "Buy",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "Sell",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "Sell",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoWidth": false,
            "children": Array [
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="USD"
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                value="USD"
              />,
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="LTC"
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                value="LTC"
              />,
            ],
            "className": "base",
            "disabled": false,
            "errorText": "",
            "floatingLabelFixed": true,
            "floatingLabelText": "Base Currency",
            "fullWidth": false,
            "multiple": false,
            "onChange": [Function],
            "value": "USD",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "USD",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "USD",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "LTC",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "LTC",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoWidth": false,
            "children": Array [
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="BTC"
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                value="BTC"
              />,
              <MenuItem
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="LTC"
                targetOrigin={
                  Object {
                    "horizontal": "left",
                    "vertical": "top",
                  }
                }
                value="LTC"
              />,
            ],
            "className": "quote",
            "disabled": false,
            "errorText": "",
            "floatingLabelFixed": true,
            "floatingLabelText": "Quote Currency",
            "fullWidth": false,
            "multiple": false,
            "onChange": [Function],
            "value": "BTC",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "BTC",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "BTC",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "LTC",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "LTC",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "amount",
            "disabled": false,
            "errorText": "",
            "floatingLabelFixed": true,
            "floatingLabelText": "Amount",
            "fullWidth": false,
            "hintText": "10.00",
            "multiLine": false,
            "name": "amount",
            "onChange": [Function],
            "rows": 1,
            "type": "text",
            "underlineShow": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "fullWidth": false,
            "label": "Submit",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": true,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "{
  \\"actions\\": [
    \\"Buy\\",
    \\"Sell\\"
  ],
  \\"current_action\\": \\"Buy\\",
  \\"curriences\\": [
    \\"USD\\",
    \\"BTC\\",
    \\"LTC\\"
  ],
  \\"base\\": \\"USD\\",
  \\"quote\\": \\"BTC\\",
  \\"amount\\": \\"\\",
  \\"actionErrorText\\": \\"\\",
  \\"baseErrorText\\": \\"\\",
  \\"quoteErrorText\\": \\"\\",
  \\"amountErrorText\\": \\"\\"
}",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "{
  \\"actions\\": [
    \\"Buy\\",
    \\"Sell\\"
  ],
  \\"current_action\\": \\"Buy\\",
  \\"curriences\\": [
    \\"USD\\",
    \\"BTC\\",
    \\"LTC\\"
  ],
  \\"base\\": \\"USD\\",
  \\"quote\\": \\"BTC\\",
  \\"amount\\": \\"\\",
  \\"actionErrorText\\": \\"\\",
  \\"baseErrorText\\": \\"\\",
  \\"quoteErrorText\\": \\"\\",
  \\"amountErrorText\\": \\"\\"
}",
            " ",
          ],
          "type": "p",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
